suitcase/Csv/__init__.py

```diff
@@ -0,0 +1,267 @@
+        '''Extracts `start` document information for formatting file_prefix.
+
+        This method checks that only one `start` document is seen and formats
+        `file_prefix` based on the contents of the `start` document.
+
+        Parameters:
+        -----------
+        doc : dict
+            RunStart document
+        '''
+
+        # raise an error if this is the second `start` document seen.
+        if self._start_found:
+            raise RuntimeError(
+                "The serializer in suitcase.tiff expects documents from one "
+                "run only. Two `start` documents where sent to it")
+        else:
+            self._start_found = True
+
+        # format self._file_prefix
+        self._templated_file_prefix = self._file_prefix.format(**doc)
+
+    def descriptor(self, doc):
+        '''Use `descriptor` doc to map stream_names to descriptor uid's.
+
+        This method usess the descriptor document information to map the
+        stream_names to descriptor uid's.
+
+        Parameters:
+        -----------
+        doc : dict
+            EventDescriptor document
+        '''
+        # extract some useful info from the doc
+        streamname = doc.get('name')
+        self._streamnames[doc['uid']] = streamname
+
+    def event_page(self, doc):
+        '''Add event page document information to a ".csv" file.
+
+        This method adds event_page document information to a ".csv" file,
+        creating it if nesecary.
+
+        .. warning::
+
+            All non 1D 'tabular' data is explicitly ignored.
+
+        .. note::
+
+            The data in Events might be structured as an Event, an EventPage,
+            or a "bulk event" (deprecated). The DocumentRouter base class takes
+            care of first transforming the other repsentations into an
+            EventPage and then routing them through here, so no further action
+            is required in this class. We can assume we will always receive an
+            EventPage.
+
+        Parameters:
+        -----------
+        doc : dict
+            EventPage document
+        '''
+        event_model.verify_filled(doc)
+        streamname = self._streamnames[doc['descriptor']]
+        valid_data = {}
+        for field in doc['data']:
+            # check that the data is 1D, if not ignore it
+            if numpy.asarray(doc['data'][field]).ndim == 1:
+                # create a file for this stream and field if required
+                if streamname not in self._files.keys():
+                    filename = (f'{self._templated_file_prefix}'
+                                f'{streamname}.csv')
+                    f = self.manager.open('stream_data', filename, 'xt')
+                    self._files[streamname] = f
+                # add the valid data to the valid_data dict
+                valid_data[field] = doc['data'][field]
+
+        if valid_data:
+
+            event_data = pandas.DataFrame(valid_data, index=doc['time'])
```

Comment:
Do you want to use `index_label` from `self._kwargs` instead of `'time'` here?

Reply:
excellent catch @mrakitin (however it is not so critical as this information is lost once it is saved to csv anyway !-) )

fixed